### PR TITLE
SUS-449 | move emoticons to the code

### PR DESCRIPTION
**Pull request by macbre**
https://wikia-inc.atlassian.net/browse/SUS-449

Fetch http://images.wikia.com/lyricwiki/images/6/67/Smile001.gif &
http://images3.wikia.nocookie.net/__cb20100822133322/lyricwiki/images/d/d8/Sad.png and serve them from cookie-less domain.

### CODE REVIEW

File: extensions/wikia/Chat2/js/emoticons.js [hunk 1]
```diff
@@ -111,12 +111,11 @@ if (typeof EmoticonMapping === 'undefined') {
 
 		// Since the values in here are processed and then cached, don't modify this directly.  Use mutators
 		// (which can invalidate the cached data such as self._regexes).
-		// TODO: fetch emoticons from nocookie domain
-		// https://wikia-inc.atlassian.net/browse/SUS-449
-		this._settings = {
-			'http://images.wikia.com/lyricwiki/images/6/67/Smile001.gif': [':)', ':-)', '(smile)'],
-			'http://images3.wikia.nocookie.net/__cb20100822133322/lyricwiki/images/d/d8/Sad.png': [':(', ':-(', ':|']
-		};
+		var emoticonsPath = window.wgExtensionsPath + '/wikia/Chat2/images/emoticons';
```
Comment (dianafa) [inline extensions/wikia/Chat2/js/emoticons.js]: why can't we move this declaration & assignment to l105?

File: extensions/wikia/Chat2/js/emoticons.js [hunk 2]
```diff
@@ -111,12 +111,11 @@ if (typeof EmoticonMapping === 'undefined') {
```
Comment (dianafa) [inline extensions/wikia/Chat2/js/emoticons.js:113]: can we update/remove above part of commented out code?

File: extensions/wikia/Chat2/js/emoticons.js [hunk 3]
```diff
@@ -111,12 +111,11 @@ if (typeof EmoticonMapping === 'undefined') {
...
-		// https://wikia-inc.atlassian.net/browse/SUS-449
-		this._settings = {
-			'http://images.wikia.com/lyricwiki/images/6/67/Smile001.gif': [':)', ':-)', '(smile)'],
-			'http://images3.wikia.nocookie.net/__cb20100822133322/lyricwiki/images/d/d8/Sad.png': [':(', ':-(', ':|']
-		};
+		var emoticonsPath = window.wgExtensionsPath + '/wikia/Chat2/images/emoticons';
+
+		this._settings = {};
+		this._settings[emoticonsPath + '/smile.png'] = [':)', ':-)', '(smile)'];
+		this._settings[emoticonsPath + '/sad.png'] = [':(', ':-(', ':|'];
```
Comment (dianafa) [inline extensions/wikia/Chat2/js/emoticons.js]: what about
```
this._settings = {
  [emoticonsPath + '/smile.png']: [':)', ':-)', '(smile)'],
  [emoticonsPath + '/sad.png']: [':(', ':-(', ':|']
};
```
?

Comment (macbre) [reply extensions/wikia/Chat2/js/emoticons.js]: JavaScript does not support dynamic key names in this syntax - you can not concatenate key names when `{}` literal is used.

File: extensions/wikia/Chat2/js/emoticons.js (hunk 1 above)
Comment (macbre) [reply extensions/wikia/Chat2/js/emoticons.js]: l105 ?

File: extensions/wikia/Chat2/js/emoticons.js (hunk 2 above)
Comment (macbre) [reply extensions/wikia/Chat2/js/emoticons.js:113]: Well, this is just an example of regexp to URL mapping :)

File: extensions/wikia/Chat2/js/emoticons.js (hunk 3 above)
Comment (dianafa) [reply extensions/wikia/Chat2/js/emoticons.js]: updated the comment, you can add `[]` to do this :)

File: extensions/wikia/Chat2/js/emoticons.js (hunk 1 above)
Comment (dianafa) [reply extensions/wikia/Chat2/js/emoticons.js]: to the top of the scope, line 105

File: extensions/wikia/Chat2/js/emoticons.js (hunk 3 above)
Comment (KockaAdmiralac) [reply extensions/wikia/Chat2/js/emoticons.js]: Isn't that a [functionality of ES6](https://stackoverflow.com/a/19837961), though? From what I've seen, ES6 code is likely to break mobile browsers (if you even support mobile browsers in chat).

Comment (macbre) [reply extensions/wikia/Chat2/js/emoticons.js]: I believe we still support browsers that are not fluent in ES6 :)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 9 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [emoticons.js#L135](https://github.com/Wikia/app/blob/703f0c530706345766b1f87225d06d4147749050/extensions/wikia/Chat2/js/emoticons.js#L135): Take the required action to fix the issue indicated by this comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1134)
1. ![MAJOR][MAJOR] [emoticons.js#L148](https://github.com/Wikia/app/blob/703f0c530706345766b1f87225d06d4147749050/extensions/wikia/Chat2/js/emoticons.js#L148): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)
1. ![MAJOR][MAJOR] [emoticons.js#L177](https://github.com/Wikia/app/blob/703f0c530706345766b1f87225d06d4147749050/extensions/wikia/Chat2/js/emoticons.js#L177): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. ![MAJOR][MAJOR] [emoticons.js#L177](https://github.com/Wikia/app/blob/703f0c530706345766b1f87225d06d4147749050/extensions/wikia/Chat2/js/emoticons.js#L177): Remove this useless assignment to local variable "keyName" [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1854)
1. ![MAJOR][MAJOR] [emoticons.js#L180](https://github.com/Wikia/app/blob/703f0c530706345766b1f87225d06d4147749050/extensions/wikia/Chat2/js/emoticons.js#L180): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. ![MAJOR][MAJOR] [emoticons.js#L180](https://github.com/Wikia/app/blob/703f0c530706345766b1f87225d06d4147749050/extensions/wikia/Chat2/js/emoticons.js#L180): Remove this useless assignment to local variable "regKeyName" [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1854)
1. ![MAJOR][MAJOR] [emoticons.js#L188](https://github.com/Wikia/app/blob/703f0c530706345766b1f87225d06d4147749050/extensions/wikia/Chat2/js/emoticons.js#L188): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. ![MAJOR][MAJOR] [emoticons.js#L190](https://github.com/Wikia/app/blob/703f0c530706345766b1f87225d06d4147749050/extensions/wikia/Chat2/js/emoticons.js#L190): Remove this useless assignment to local variable "regexString" [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1854)
1. ![INFO][INFO] [emoticons.js#L135](https://github.com/Wikia/app/blob/703f0c530706345766b1f87225d06d4147749050/extensions/wikia/Chat2/js/emoticons.js#L135): Complete the task associated to this TODO comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/Chat2/js/emoticons.js (hunk 1 above)
Comment (macbre) [reply extensions/wikia/Chat2/js/emoticons.js]: Good point - moved top in aa6ac9536f5

**Comment by WikiaTech** [comment]
SonarQube analysis reported 9 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [emoticons.js#L135](https://github.com/Wikia/app/blob/aa6ac9536f5546b535f7df20095eab11bb50dfa6/extensions/wikia/Chat2/js/emoticons.js#L135): Take the required action to fix the issue indicated by this comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1134)
1. ![MAJOR][MAJOR] [emoticons.js#L148](https://github.com/Wikia/app/blob/aa6ac9536f5546b535f7df20095eab11bb50dfa6/extensions/wikia/Chat2/js/emoticons.js#L148): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)
1. ![MAJOR][MAJOR] [emoticons.js#L177](https://github.com/Wikia/app/blob/aa6ac9536f5546b535f7df20095eab11bb50dfa6/extensions/wikia/Chat2/js/emoticons.js#L177): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. ![MAJOR][MAJOR] [emoticons.js#L177](https://github.com/Wikia/app/blob/aa6ac9536f5546b535f7df20095eab11bb50dfa6/extensions/wikia/Chat2/js/emoticons.js#L177): Remove this useless assignment to local variable "keyName" [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1854)
1. ![MAJOR][MAJOR] [emoticons.js#L180](https://github.com/Wikia/app/blob/aa6ac9536f5546b535f7df20095eab11bb50dfa6/extensions/wikia/Chat2/js/emoticons.js#L180): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. ![MAJOR][MAJOR] [emoticons.js#L180](https://github.com/Wikia/app/blob/aa6ac9536f5546b535f7df20095eab11bb50dfa6/extensions/wikia/Chat2/js/emoticons.js#L180): Remove this useless assignment to local variable "regKeyName" [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1854)
1. ![MAJOR][MAJOR] [emoticons.js#L188](https://github.com/Wikia/app/blob/aa6ac9536f5546b535f7df20095eab11bb50dfa6/extensions/wikia/Chat2/js/emoticons.js#L188): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. ![MAJOR][MAJOR] [emoticons.js#L190](https://github.com/Wikia/app/blob/aa6ac9536f5546b535f7df20095eab11bb50dfa6/extensions/wikia/Chat2/js/emoticons.js#L190): Remove this useless assignment to local variable "regexString" [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1854)
1. ![INFO][INFO] [emoticons.js#L135](https://github.com/Wikia/app/blob/aa6ac9536f5546b535f7df20095eab11bb50dfa6/extensions/wikia/Chat2/js/emoticons.js#L135): Complete the task associated to this TODO comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by sqreek — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 9 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [emoticons.js#L135](https://github.com/Wikia/app/blob/6d7445a59d009d80f2ae79dbd53180d7027ba45e/extensions/wikia/Chat2/js/emoticons.js#L135): Take the required action to fix the issue indicated by this comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1134)
1. ![MAJOR][MAJOR] [emoticons.js#L148](https://github.com/Wikia/app/blob/6d7445a59d009d80f2ae79dbd53180d7027ba45e/extensions/wikia/Chat2/js/emoticons.js#L148): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)
1. ![MAJOR][MAJOR] [emoticons.js#L177](https://github.com/Wikia/app/blob/6d7445a59d009d80f2ae79dbd53180d7027ba45e/extensions/wikia/Chat2/js/emoticons.js#L177): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. ![MAJOR][MAJOR] [emoticons.js#L177](https://github.com/Wikia/app/blob/6d7445a59d009d80f2ae79dbd53180d7027ba45e/extensions/wikia/Chat2/js/emoticons.js#L177): Remove this useless assignment to local variable "keyName" [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1854)
1. ![MAJOR][MAJOR] [emoticons.js#L180](https://github.com/Wikia/app/blob/6d7445a59d009d80f2ae79dbd53180d7027ba45e/extensions/wikia/Chat2/js/emoticons.js#L180): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. ![MAJOR][MAJOR] [emoticons.js#L180](https://github.com/Wikia/app/blob/6d7445a59d009d80f2ae79dbd53180d7027ba45e/extensions/wikia/Chat2/js/emoticons.js#L180): Remove this useless assignment to local variable "regKeyName" [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1854)
1. ![MAJOR][MAJOR] [emoticons.js#L188](https://github.com/Wikia/app/blob/6d7445a59d009d80f2ae79dbd53180d7027ba45e/extensions/wikia/Chat2/js/emoticons.js#L188): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. ![MAJOR][MAJOR] [emoticons.js#L190](https://github.com/Wikia/app/blob/6d7445a59d009d80f2ae79dbd53180d7027ba45e/extensions/wikia/Chat2/js/emoticons.js#L190): Remove this useless assignment to local variable "regexString" [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1854)
1. ![INFO][INFO] [emoticons.js#L135](https://github.com/Wikia/app/blob/6d7445a59d009d80f2ae79dbd53180d7027ba45e/extensions/wikia/Chat2/js/emoticons.js#L135): Complete the task associated to this TODO comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**